File: cs/Iris/ExpressionElement/Instruction/MenuInstruction.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Iris.ParserUtils;

namespace Iris.ExpressionElement.Instruction
{
    public class MenuInstruction : InstructionExpression
    {
        //menu("Oui je veux bien", "Non merci, ca ira", "Avec plaisir"){
        //  case 0:
        //  case 2:
        //      open;
        //      msg "Tres bien suivez moi";
        //      close;
        //      break;
        //
        //  case 1:
        //      open;
        //      msg "Oh mais pourquoi etes vous si serieux ?";
        //      close;
        //}

        public List<string> choices;
        public List<SwitchCaseExpression> cases;

        public bool hasDefaultCase = false;

        public List<int> intCases;

        public MenuInstruction()
        {

        }

        public void addChoice(string choice)
        {
            if (choices == null)
            {
                choices = new List<string>();
            }

            choices.Add(choice);
        }

        public List<string> declaredVariables()
        {
            return null;
        }

        public void addCase(SwitchCaseExpression ce)
        {
            if (cases == null)
            {
                cases = new List<SwitchCaseExpression>();
            }

            cases.Add(ce);

            if (ce is DefaultExpression)
            {
                hasDefaultCase = true;
                return;
            }

            if (ce is CaseExpression)
            {
                SwitchCaseExpression tmp = ce;

                while (tmp != null)
                {
                    if (tmp is DefaultExpression)
                    {
                        hasDefaultCase = true;
                        return;
                    }
                    else
                    {
                        CaseExpression caseExp = (CaseExpression)tmp;

                        if (intCases == null)
                        {
                            intCases = new List<int>();
                        }

                        intCases.Add(caseExp.literal.value.intValue);
                        tmp = caseExp.next;
                    }
                }
            }
        }
    }
}

File: java/Iris/src/com/rokuan/iris/instruction/MenuInstruction.java
package com.rokuan.iris.instruction;

import java.util.ArrayList;

import com.rokuan.iris.expression.CaseExpression;
import com.rokuan.iris.expression.DefaultExpression;
import com.rokuan.iris.expression.SwitchCaseExpression;

public class MenuInstruction extends InstructionExpression
{
	//menu("Oui je veux bien", "Non merci, ca ira", "Avec plaisir"){
	//  case 0:
	//  case 2:
	//      open;
	//      msg "Tres bien suivez moi";
	//      close;
	//      break;
	//
	//  case 1:
	//      open;
	//      msg "Oh mais pourquoi etes vous si serieux ?";
	//      close;
	//}

	public ArrayList<String> choices;
	public ArrayList<SwitchCaseExpression> cases;

	public boolean hasDefaultCase = false;

	public ArrayList<Integer> intCases;

	public MenuInstruction()
	{

	}

	public void addChoice(String choice)
	{
		if (choices == null)
		{
			choices = new ArrayList<String>();
		}

		choices.add(choice);
	}

	public ArrayList<String> declaredVariables()
	{
		return null;
	}

	public void addCase(SwitchCaseExpression ce)
	{
		if (cases == null)
		{
			cases = new ArrayList<SwitchCaseExpression>();
		}

		cases.add(ce);

		if (ce instanceof DefaultExpression)
		{
			hasDefaultCase = true;
			return;
		}

		if (ce instanceof CaseExpression)
		{
			SwitchCaseExpression tmp = ce;

			while (tmp != null)
			{
				if (tmp instanceof DefaultExpression)
				{
					hasDefaultCase = true;
					return;
				}
				else
				{
					CaseExpression caseExp = (CaseExpression)tmp;

					if (intCases == null)
					{
						intCases = new ArrayList<Integer>();
					}

					intCases.add(caseExp.literal.value.intValue);
					tmp = caseExp.next;
				}
			}
		}
	}
}
